executable-spec/src/Cardano/Ledger/Spec/STS/Update/Ideation.hs

```diff
@@ -191,18 +193,25 @@ instance
 
   sigGen
     _traceGenEnv
-    Env{ k, currentSlot, participants }
+    Env{ k, currentSlot, asips, participants }
     St{ wssips, wrsips, subsips } =
       case Set.toList $ range $ stableCommits ◁ subsips of
         [] ->
           -- There are no stable commits, so we can only generate a submission.
           submissionGen
         xs ->
-          -- We use a 50/50 submission to revelation ratio. This can be changed
-          -- if necessary.
-          QC.frequency [ (1, submissionGen)
-                       , (1, revelationGen xs)
-                       ]
+          if Set.empty == dom (asips ▷>= currentSlot)
+            then
+              -- We use a 50/50 submission to revelation ratio. This can be changed
+              -- if necessary.
+              QC.frequency [ (1, submissionGen)
+                           , (1, revelationGen xs)
+                           ]
+            else
+              QC.frequency [ (1, submissionGen)
+                           , (1, revelationGen xs)
+                           , (998, voteGen asips participants currentSlot)
```

Comment:
This constant seems huge. Do we need it to be so large, and if so we might want to explain why.

Reply:
Done. OK